### DISCUSSION TITLE
swift pareser in go

### DISCUSSION THREAD

**Discussion by OmarEldeeb98** [antlr/antlr4#4676] [Q&A] [answered]
I am trying to generate swift5 parser in golang but after generating it 
swift5_parser.go and swift5_lexer.go have many error 
i have downloaded Swift5Lexer.g4 and Swift5Parser.g4 and ran
`antlr4 -Dlanguage=Go -o  output-directory Swift5Lexer.g4 Swift5Parser.g4`
one of errors in swift5_parser.go in `func (p *Swift5Parser) Statements() (localctx IStatementsContext)`
all fucctions of p are undefined

**Comment by kaby76** [comment, marked as answer]
* This grammar is in grammars-g4 https://github.com/antlr/grammars-v4/tree/8e1413efb04f0bf97aea5674cf8d18e970972934/swift/swift5.
* There is no Go port of the grammar. https://github.com/antlr/grammars-v4/blob/8e1413efb04f0bf97aea5674cf8d18e970972934/swift/swift5/desc.xml#L4
* It takes a lot of time and effort to write a port for a grammar, which is why it doesn't exist.

**Reply by OmarEldeeb98** [reply]
i have generated kotlin parser in go and it works well, in spite that target in `desc.xml` in case of kotlin is empty, https://github.com/antlr/grammars-v4/blob/master/kotlin/kotlin/desc.xml
why is that different to swift5 parser ?
thank you

**Reply by kaby76** [reply]
Thanks for letting me know about the kotlin grammar desc.xml. I'm not sure why there are no targets specified in the [desc.xml](https://github.com/antlr/grammars-v4/blob/8e1413efb04f0bf97aea5674cf8d18e970972934/kotlin/kotlin/desc.xml#L3). That seems like an error. But, there are some errors for [examples/script/preable_nl.kts](https://github.com/antlr/grammars-v4/blob/8e1413efb04f0bf97aea5674cf8d18e970972934/kotlin/kotlin/examples/script/preamble_nl.kts), probably because there is no newline on line 6. Maybe that is why I removed the targets.
